src/einsteinpy/coordinates/conversion.py

```diff
@@ -71,380 +78,120 @@ def values(self):
 
         return self.t_si, self.x_si, self.y_si, self.z_si
 
-    def convert_spherical(self, **kwargs):
+
+class CartesianConversion(CoordinateConversionBase):
+    """
+    Class for conversion to and from Cartesian Coordinates in SI units
+    """
+
+    def convert_spherical(self) -> Tuple[float, float, float, float]:
         """
         Converts to Spherical Polar Coordinates
 
-        Other Parameters
-        ----------------
-        **kwargs : dict
-            Keyword Arguments
-
         Returns
         -------
         tuple
-            4-Tuple or 7-Tuple, containing the components in \
-            Spherical Polar Coordinates
-
+            4-Tuple containing the components in Spherical Polar Coordinates
         """
-        return cartesian_to_spherical_fast(
-            self.t_si,
-            self.x_si,
-            self.y_si,
-            self.z_si,
-            self.v_x_si,
-            self.v_y_si,
-            self.v_z_si,
-            self._velocities_provided,
-        )
+        return cartesian_to_spherical_fast(*self.values())
 
-    def convert_bl(self, **kwargs):
+    def convert_bl(self, M: float, a: float) -> Tuple[float, float, float, float]:
         """
         Converts to Boyer-Lindquist Coordinates
 
         Parameters
         ----------
-        **kwargs : dict
-            Keyword Arguments
-            Expects two arguments, ``M and ``a``, as described below
-
-        Other Parameters
-        ----------------
         M : float
-            Mass of the gravitating body, \
-            around which, spacetime has been defined
+            Mass of the gravitating body
         a : float
-            Spin Parameter of the gravitating body, \
-            around which, spacetime has been defined
+            Spin Parameter of the gravitating body
 
         Returns
         -------
         tuple
-            4-Tuple or 7-Tuple, containing the components in \
-            Boyer-Lindquist Coordinates
-
-        Raises
-        ------
-        KeyError
-            If ``kwargs`` does not contain both ``M`` \
-            and ``a`` as keyword arguments
-
+            4-Tuple containing the components in Boyer-Lindquist Coordinates
         """
-        try:
-            M, a = kwargs["M"], kwargs["a"]
-        except KeyError:
-            raise KeyError(
-                "Two keyword arguments are expected: Mass, 'M' and Spin Parameter, 'a'."
-            )
-
         alpha = BaseMetric.alpha(M=M, a=a)
-
-        return cartesian_to_bl_fast(
-            self.t_si,
-            self.x_si,
-            self.y_si,
-            self.z_si,
-            alpha,
-            self.v_x_si,
-            self.v_y_si,
-            self.v_z_si,
-            self._velocities_provided,
-        )
+        return cartesian_to_bl_fast(*self.values(), alpha)
```

Comment:
See the previous comment about `utils.py`. This parameter ordering is incorrect.